### ISSUE TITLE
[BUG] 

### ISSUE THREAD

**Issue by Danthar** [AzureAD/microsoft-identity-web#2576]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

2.15.3

### Web app

Sign-in users and call web APIs

### Web API

Not Applicable

### Token cache serialization

In-memory caches

### Description

When using the `AddMicrosoftIdentityWebApp` extension method as defined in the `AppBuilderExtension.cs` file. The default `RedirectToIdentityProvider` handler defined on the Notification options for the `OpenIdConnectAuthenticationOptions` object. Is bugged. 

This is due to how the `OpenIdConnectMessage` type works.

Im using the `Microsoft.IdentityModel.Protocols.OpenIdConnect` package version  7.0.3. And im not sure if previous versions with this package also have this issue.

The following problem is at hand.

This is currently the code:

```

 RedirectToIdentityProvider = (context) =>
                    {
                        var loginHint = context.ProtocolMessage.GetParameter(OpenIdConnectParameterNames.LoginHint);
                        if (!string.IsNullOrWhiteSpace(loginHint))
                        {
                            context.ProtocolMessage.LoginHint = loginHint;

                            context.ProtocolMessage.SetParameter(Constants.XAnchorMailbox, $"{Constants.Upn}:{loginHint}");
                            // delete the login_hint from the Properties when we are done otherwise
                            // it will take up extra space in the cookie.
                            context.ProtocolMessage.Parameters.Remove(OpenIdConnectParameterNames.LoginHint);
                        }

                        var domainHint = context.ProtocolMessage.GetParameter(OpenIdConnectParameterNames.DomainHint);
                        if (!string.IsNullOrWhiteSpace(domainHint))
                        {
                            context.ProtocolMessage.DomainHint = domainHint;

                            // delete the domain_hint from the Properties when we are done otherwise
                            // it will take up extra space in the cookie.
                            context.ProtocolMessage.Parameters.Remove(OpenIdConnectParameterNames.DomainHint);
                        }
                        context.ProtocolMessage.SetParameter(ClaimConstants.ClientInfo, Constants.One);
                        context.ProtocolMessage.SetParameter(Constants.TelemetryHeaderKey, IdHelper.CreateTelemetryInfo());
                        return Task.CompletedTask;
                    },
                 
```

Its logic is simple. But wrong. 
It attempts to read the loginhint and domainhint parameters from the protocol message, and then sets the LoginHint and DomainHint Property on the Protocol message.

However those properties are nothing more then wrappers around GetParameter and SetParameter calls. 
Thus 
1: Nothing actually changes here
2: The Parameters.Remove call actually deletes those values again, making it a unintentionally destructive operation.

Suggested changes:
Only set the XAnchorMailbox parameter and get rid of all other logic surrounding LoginHint and DomainHint in this code. 

### Reproduction steps

Pick one of the quickstarts, attach a debugger, and have at it. 

### Error message

Not getting any specific errors. Other then that currently out of the box the msgraph integration does not work. 
Meaning the standard snippet from the Quickstart tutorial does not work:

```
  factory.Services
                 .AddMicrosoftGraph()
                 .AddInMemoryTokenCaches();
             factory.Build();

```

Yes the graph client is resolvable, but any call will result in the following error:

**No account or login hint was passed to the AcquireTokenSilent call.** 

And im not sure if the above bugreport has anything to do with it. But this error is what triggered me to dig into this. 

### Relevant code snippets

```csharp
See description
```

### Expected behavior

I expect the default implementation to leave the login and domainhint intact. 
And i expect that when using the standard samples from the docs that the MSGraph integration works. (but that might be for another issue)

**Comment by jennyf19** [comment]
@Danthar Where did you find this [sample](https://github.com/AzureADQuickStarts/AppModelv2-WebApp-OpenIDConnect-DotNet/blob/master/AppModelv2-WebApp-OpenIDConnect-DotNet/Startup.cs)? is it linked from a doc? can you share the doc, if so.

**Comment by Danthar** [comment]
Yes its linked from here: https://learn.microsoft.com/en-us/entra/identity-platform/sample-v2-code?tabs=apptype
Also defined here: https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-web-app-aspnet-sign-in

![afbeelding](https://github.com/AzureAD/microsoft-identity-web/assets/986519/be5118ac-19a5-4954-a48e-27b75d59ed06)

**Comment by Danthar** [comment]
Note that i found out why the msgraph integration did not work. And reported this in a seperate issue:
https://github.com/AzureAD/microsoft-identity-web/issues/2586
